astropy/modeling/core.py

```diff
@@ -2315,6 +2407,113 @@ def _custom_model_wrapper(func, fit_deriv=None):
     return type(model_name, (FittableModel,), members)
 
 
+def render_model(model, arr=None, coords=None, copy=True):
+    """
+    Evaluates a model on an input array. Evaluation is limited to
+    a bounding box if the `Model.bounding_box` attribute is set.
+
+    Parameters
+    ----------
+    model : `Model`
+        Model to be evaluated.
+    arr : `numpy.ndarray`, optional
+        Array on which the model is evaluated.
+    coords : array-like, optional
+        Coordinate arrays mapping to ``array``, such that 
+        ``arr[coords] == arr``.
+    copy : bool, optional
+        Default is ``True``. If ``False``, ``arr`` will be modified in place. 
+
+    Returns
+    -------
+    array : `numpy.ndarray`
+        The model evaluated on the input ``arr`` or a new array from ``coords``.  
+        If ``arr`` and ``coords`` are both `None`, the returned array is 
+        limited to the `Model.bounding_box` limits. If 
+        `Model.bounding_box` is `None`, ``arr`` or ``coords`` must be passed.
+    """
+
+    bbox = model.bounding_box
+
+    if (coords is None) & (arr is None) & (bbox is None):
+        raise AssertionError('If no bounding_box is set, coords or arr must be input.')
+
+    # for consistent indexing
+    if model.n_inputs == 1:
+        if coords is not None:
+            coords = [coords]
+        if bbox is not None:
+            bbox = [bbox]
+
+    if arr is not None:
+        # Check dimensions match model
+        assert arr.ndim == model.n_inputs
+        # Copy
+        if copy:
+            arr = arr.copy()
+
+    if coords is not None:
+        # Will fail if coords are inconsisent
+        coords = np.array(coords)
+
+        # Check dimensions match model
+        assert len(coords) == model.n_inputs
+
+        # Check dimensions match arr
+        if arr is not None:
+            assert coords[0].shape == arr.shape
+
+        # If no input arr, make one
+        else:
+            arr = np.zeros(coords[0].shape)
+
+    if bbox is not None:
+
+        # assure position is at center pixel
+        pd = pos, delta = np.array([(np.mean(bb), np.ceil((bb[1] - bb[0]) / 2))
+                                    for bb in bbox]).astype(int).T
+
+        if coords is not None:
```

Comment:
I think there's no coding style guideline for this, so it's up to you @patti.

But I'd suggest you use fewer empty lines.
Specifically I find the empty lines before else clauses make it harder to see where one if-else block starts and ends.

You're using this formatting:
```python
if condition:

    statements

else:

    statements

statements
```

I'd suggest (and I think also most code in Astropy is formatted like this and it means that more code fits on one page and IMO is even more readable):
```python
if condition:
    statements
else:
    statements

statements
```